### ISSUE TITLE
Version 1.0.0 Error: java.lang.SecurityException: Starting FGS with type microphone callerApp=ProcessRecord targetSDK=34 requires permissions

### ISSUE THREAD

**Issue by dev3whistlebiz** [twilio/twilio-voice-react-native#349]
# Issue
Inbound call on android is crashing the app.

## Pre-submission Checklist
- [x] I have verified that the issue occurs with the latest release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-voice-react-native/blob/main/CHANGELOG.md).
- [x] I reviewed the [Common Issues](https://github.com/twilio/twilio-voice-react-native/blob/main/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
- [x] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

## Description 
A clear and concise description of what the issue is.
Using expo with react native. On inbound call android is not showing the call and the application is crashing.

## Reproduction Steps
1. Open application.
2. Background the application.
3. Receive a call.
4. Application stopped working, error was thrown.

## Expected Behavior

1. Open application.
2. Background the application.
3. Receive a call.
4. Application showed notification.
5. Call can be accepted/declined.

## Actual Behavior
What actually happens.

1. Open application.
2. Background the application.
3. Receive a call.
4. Application stopped working.
5. In terminal error is shown as below:
`
Your app just crashed. See the error below.

java.lang.SecurityException: Starting FGS with type microphone callerApp=ProcessRecord ... targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MICROPHONE] any of the permissions allOf=false [android.permission.CAPTURE_AUDIO_HOTWORD, android.permission.CAPTURE_AUDIO_OUTPUT, android.permission.CAPTURE_MEDIA_OUTPUT, android.permission.CAPTURE_TUNER_AUDIO_INPUT, android.permission.CAPTURE_VOICE_COMMUNICATION_OUTPUT, android.permission.RECORD_AUDIO]  and the app must be in the eligible state/exemptions to access the foreground only permission

  android.os.Parcel.createExceptionOrNull(Parcel.java:3182)
  android.os.Parcel.createException(Parcel.java:3166)
  android.os.Parcel.readException(Parcel.java:3149)
  android.os.Parcel.readException(Parcel.java:3091)
  android.app.IActivityManager$Stub$Proxy.setServiceForeground(IActivityManager.java:6861)
  android.app.Service.startForeground(Service.java:862)
  com.twiliovoicereactnative.VoiceService.foregroundNotification(VoiceService.java:317)
  com.twiliovoicereactnative.VoiceService.createOrReplaceNotification(VoiceService.java:307)
  com.twiliovoicereactnative.VoiceService.incomingCall(VoiceService.java:166)
  com.twiliovoicereactnative.VoiceService.-$$Nest$mincomingCall(Unknown Source:0)
  com.twiliovoicereactnative.VoiceService$VoiceServiceAPI.incomingCall(VoiceService.java:73)
  com.twiliovoicereactnative.VoiceFirebaseMessagingService.onCallInvite(VoiceFirebaseMessagingService.java:65)
  com.twilio.voice.CallInviteProxy.lambda$onCallInvite$0$com-twilio-voice-CallInviteProxy(CallInviteProxy.java:82)     
  com.twilio.voice.CallInviteProxy$$ExternalSyntheticLambda2.run(Unknown Source:4)
  android.os.Handler.handleCallback(Handler.java:959)
  android.os.Handler.dispatchMessage(Handler.java:100)
  android.os.Looper.loopOnce(Looper.java:232)
  android.os.Looper.loop(Looper.java:317)
  android.app.ActivityThread.main(ActivityThread.java:8501)
  java.lang.reflect.Method.invoke(Native Method)
  com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
  com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
Caused by android.os.RemoteException: Remote stack trace:
        at com.android.server.am.ActiveServices.validateForegroundServiceType(ActiveServices.java:2685)
        at com.android.server.am.ActiveServices.setServiceForegroundInnerLocked(ActiveServices.java:2375)
        at com.android.server.am.ActiveServices.setServiceForegroundLocked(ActiveServices.java:1704)
        at com.android.server.am.ActivityManagerService.setServiceForeground(ActivityManagerService.java:13684)        
        at android.app.IActivityManager$Stub.onTransact(IActivityManager.java:3430)
› Stopped server
`

## Reproduction Frequency
Is the reproducibility of the issue deterministic? If not, what percentage of the time does the issue occur? In how many attempts was the issue observed?

100%

## Software and Device Information
**Please complete the following information.**
 - Device: Pixel 7
 - OS: Android

## Additional Context

Manifest.xml has all the permissions that are specified in the error. Those permissions are granted. I am aware of targetSdk 34-th permissions updates regarding `android:foregroundServiceType` and I set it to `microphone`.

**Comment by dev3whistlebiz** [comment]
Any updates?

**Comment by dev3whistlebiz** [comment]
Hey, any updates on this one?

**Comment by mhuynh5757** [comment]
Hi @dev3whistlebiz sorry for the delayed response. Is this an application of your own where you're implementing our SDK, or are you using the Test App that's part of this repo? If you could reproduce this in our Test App (located in the `test/app` folder) that would help us tremendously.

**Comment by dev3whistlebiz** [comment]
Hello @mhuynh5757. Thanks for the response. I am implementing yout SDK in my application which, unfortunately I cannot share details of but FYI all the permissions stated in error are defined and accepted (by me on mobile device while testing).

**Comment by mhuynh5757** [comment]
Hi @dev3whistlebiz could you try to reproduce this with the Test App that's a part of this repo?

**Comment by dev3whistlebiz** [comment]
Hello @mhuynh5757, I might try later but for our information I am using Expo. Thanks for response, I am looking forward for any updates!

**Comment by dev3whistlebiz** [comment]
Hello, any updates on this one?

**Comment by afalls-twilio** [comment]
@dev3whistlebiz for now, set your target SDK to 33, we will take a look at this and get back to you

**Comment by afalls-twilio** [comment]
@dev3whistlebiz Our current SDK is targeting API 33, if you want to use API 34, the way to handle this is to add the following to your application's manifest..
```  <uses-permission android:name="android.permission.FOREGROUND_SERVICE_MICROPHONE"/>```
and it should work, I tested it personally.

**Comment by dev3whistlebiz** [comment]
Hello, @afalls-twilio, thanks for the response. Unfortunately, Expo 50 uses API 34, so it is not possible to use 33-rd one.  I've tried `<uses-permission android:name="android.permission.FOREGROUND_SERVICE_MICROPHONE"/>` before and it was not working as well, no changes. Do you know when it can be possibly resolved? Any feedback would be very much appreciated!!

**Comment by mhuynh5757** [comment]
The fix should be complete and merged to main, however we don't have a timeline for the next release. If you are willing to try installing the library through source, it may fix your issue.

**Comment by sshodges** [comment]
@mhuynh5757 any idea when this is going to be released? getting the same error (Expo, API 34)

**Comment by sshodges** [comment]
@dev3whistlebiz did you manage to find a work around?

**Comment by sshodges** [comment]
@dev3whistlebiz  this doesn't seem to be working for me, have you written a custom Foreground Service? i was targeting the `com.twiliovoicereactnative.VoiceService`.  Seems like the manifest has the `FOREGROUND_SERVICE_MICROPHONE` permission, but still same error. I guess it could be that the SDK still targeting API 33

**Comment by Jobaer-hassan-orangetoolz** [comment]
is there any update on this? i am facing the same issue on this.
1. open the application
2. background it
3. call initiated
4. app crashes and throw error like this 


@mhuynh5757 @afalls-twilio  do you guys have some timeline when the next release on live targeting api 34?

**Comment by liemdo** [comment]
We also have this error which is affecting many users.

**Comment by mhuynh5757** [comment]
Hi all, apologies for the lack of communication. We are still working on creating a release that includes this fix. Thanks for your patience.

**Comment by disolaterX** [comment]
> Hi all, apologies for the lack of communication. We are still working on creating a release that includes this fix. Thanks for your patience.

have you figured out the exact way to fix this yet ? If yes then can you share it on android level what is the issue.

Thanks in advance.

**Comment by sshodges** [comment]
I am still looking for a fix as well, can't release the android version of app until this is fixed

**Comment by Victor-Sun** [comment]
Hello I am receiving this issue as well. Google just released a new requirement that apps must target API 34 or we will not be able to publish apps on google play. I have tried patching the package but it doesn't work either.

**Comment by vlassius** [comment]
java.lang.RuntimeException: Unable to start service com.mais.appmais.serviceAssistente.ServiceAssistente@2a04ad9 with Intent { act=start cmp=com.mais.appmais/.serviceAssistente.ServiceAssistente }: java.lang.SecurityException: Starting FGS with type microphone callerApp=ProcessRecord{2c3308b 13436:com.mais.appmais/u0a190} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MICROPHONE] any of the permissions allOf=false [android.permission.CAPTURE_AUDIO_HOTWORD, android.permission.CAPTURE_AUDIO_OUTPUT, android.permission.CAPTURE_MEDIA_OUTPUT, android.permission.CAPTURE_TUNER_AUDIO_INPUT, android.permission.CAPTURE_VOICE_COMMUNICATION_OUTPUT, android.permission.RECORD_AUDIO]  and the app must be in the eligible state/exemptions to access the foreground only permission

**Comment by Victor-Sun** [comment]
> Hi all, thanks for bringing this to our attention. I'll discuss creating a release with the team.
> 
> @Victor-Sun do you experience that error even after following the new Android integration documentation?

Hi @mhuynh5757 Yes I receive the same error

**Comment by liemdo** [comment]
Is there a workaround for this issue? Android phones crash because of this error.

**Comment by mhuynh5757** [comment]
Hey @Victor-Sun I tested the release candidate `1.1.0-rc1` on Android API 34 and I'm seeing the error that you're describing. If I'm understanding you correctly, it's not the same issue as the original one posted in this issue, correct? If so, please feel free to create another issue.

**Comment by mhuynh5757** [comment]
@liemdo are you also referring to the permissions error described originally in this Github Issue?

**Comment by mhuynh5757** [comment]
Hey all, version `1.1.0` of the SDK is now live! Please update at your earliest convenience. It should solve this issue, especially regarding Android API 34 support. I'm going to close this issue for now, if you have further need of assistance please open a new issue.

**Comment by beeeshaaal** [comment]
For the future reference:
Any one of these permissions need to be granted before starting the foreground service.
`[android.permission.CAPTURE_AUDIO_HOTWORD, android.permission.CAPTURE_AUDIO_OUTPUT, android.permission.CAPTURE_MEDIA_OUTPUT, android.permission.CAPTURE_TUNER_AUDIO_INPUT, android.permission.CAPTURE_VOICE_COMMUNICATION_OUTPUT, android.permission.RECORD_AUDIO]`
